torch/testing/_internal/common_nn.py

```diff
@@ -4905,120 +4989,124 @@ def _do_test(self, test_case, module, input):
 
             module_ip = self.constructor(*self.constructor_args, inplace=True)
 
-            input_version = input._version
+            input_versions = tuple(t._version for t in input_tuple)
             with freeze_rng_state():
-                output = module(input)
-            test_case.assertEqual(input._version, input_version)
+                output = module(*input_tuple)
+            new_input_versions = tuple(t._version for t in input_tuple)
+            test_case.assertEqual(input_versions, new_input_versions)
 
-            input_ip = deepcopy(input)
-            input_ip_clone = input_ip.clone()
+            input_ip = deepcopy(input_tuple)
+            input_ip_clone = tuple(t.clone() for t in input_ip)
             with freeze_rng_state():
-                output_ip = module_ip(input_ip_clone)
-            test_case.assertNotEqual(input_ip_clone._version, input_version)
+                output_ip = module_ip(*input_ip_clone)
+            input_ip_clone_versions = tuple(t._version for t in input_ip_clone)
+            test_case.assertNotEqual(input_ip_clone_versions, input_versions)
             test_case.assertEqual(output, output_ip)
             grad = output.data.clone().normal_()
-            input.grad.data.zero_()
+            for t in input_tuple:
+                t.grad.data.zero_()
             output.backward(grad)
             output_ip.backward(grad)
-            test_case.assertEqual(input.grad, input_ip.grad)
+            for t1, t2 in zip(input_tuple, input_ip):
+                test_case.assertEqual(t1.grad, t2.grad)
 
-        if isinstance(input, torch.LongTensor) and TEST_CUDA:
+        if any(isinstance(t, torch.LongTensor) for t in input_tuple) and TEST_CUDA:
             # check that cuda() moves module parameters to correct GPU device,
             # and that float() casts parameters correctly
-
-            input = input.cuda()
+            input_tuple = tuple(t.cuda() for t in input_tuple)
             module.float().cuda()
-            module(input)
+            module(*input_tuple)
             for p in module.parameters():
                 test_case.assertIsInstance(p, torch.cuda.FloatTensor)
                 test_case.assertEqual(p.get_device(), 0)
 
             if torch.cuda.device_count() > 1:
-                input = input.cuda(1)
+                input_tuple = tuple(t.cuda(1) for t in input_tuple)
```

Comment:
For PyTorch C++ API Parity -- are you planning on adding a Parity test for nn.LSTM at some point?

This PR extends NewModuleTest to accept tuples of floating-point tensors as input. However, supporting something like nn.LSTM would require supporting more than just tuples of floating-point tensors. LSTM takes in (input, (hidden_x, cell_x)) arguments. It might be worth sitting down and thinking about how to generalize NewModuleTest if we want to check parity for nn.LSTM in the future.

Reply:
Yes, this will be done after adding `input_dict`